apps/web/pages/api/user/avatar.ts

```diff
@@ -12,22 +12,35 @@ const querySchema = z
   .object({
     username: z.string(),
     teamname: z.string(),
+    /**
+     * Allow fetching avatar of a particular organization
+     * Avatars being public, we need not worry about others accessing it.
+     */
+    orgId: z.string().transform((s) => Number(s)),
   })
   .partial();
 
 async function getIdentityData(req: NextApiRequest) {
-  const { username, teamname } = querySchema.parse(req.query);
+  const { username, teamname, orgId } = querySchema.parse(req.query);
   const { currentOrgDomain, isValidOrgDomain } = orgDomainConfig(req.headers.host ?? "");
+
   const org = isValidOrgDomain ? currentOrgDomain : null;
+
+  const orgQuery = orgId
+    ? {
+        id: orgId,
+      }
+    : isValidOrgDomain
+    ? {
+        slug: currentOrgDomain,
```

Comment:
It could happen that the organization is unpublished, having to match the slug with `metadata.requestedSlug`